### ISSUE TITLE
How i can compile

### ISSUE THREAD

**Issue by inukaze** [protoman/rockbot#115]
Hi there, where are the instructions for how compile the source code because i do the follow :

```
git clone https://github.com/protoman/rockbot
cd rockbot
./fetch.sh
make
```
make: *** No se especificó ningún objetivo y no se encontró ningún makefile.  Alto.

ok i try again :

```
$ git clone https://github.com/protoman/rockbot
$ cd rockbot
$ ./fetch.sh
$ qmake rockbot.prx
$ make
```
g++ -m64 -Wl,-O1 -o rockbot    -L/usr/X11R6/lib64 -lQt5Gui -lQt5Core -lGL -lpthread 
/usr/lib/gcc/x86_64-linux-gnu/6/../../../x86_64-linux-gnu/Scrt1.o: En la función `_start':
(.text+0x20): referencia a `main' sin definir
collect2: error: ld returned 1 exit status
Makefile:173: fallo en las instrucciones para el objetivo 'rockbot'
make: *** [rockbot] Error 1

**Comment by protoman** [comment]
OK, before the build you will need Qt4 or Qt5, as I used qmake for the
builds.

The other dependencies also depends on the platform you are building for -
let me know so I can give further assistance - but you will need SDL,
SDL_ttf, SDL_mixer, SDL_image, SDL_gfx.

Open rockbot/Rockbot.pro in a text editor.
Uncomment the CONFIG for the platform you are building for, example, in
Windows, leave #CONFIG += win32 uncommented by removing #.
Comment all remaining platform CONFIGs.

cd rockbot
qmake
make

--
Iuri Fiedoruk, Software Developer

**Comment by inukaze** [comment]
Ok here the commands i use to compile under GNU/Linux : 

```
git clone https://github.com/protoman/rockbot
cd rockbot
./fetch.sh
sed -i '8d' ./Rockbot.pro
sed -i '17d' ./Rockbot.pro
sed -i '13 s/^.//' ./Rockbot.pro
qmake
make
```

Now i had compile the game i just need move the files : 

```
mkdir -p ../rockbot-game
mv ./rockbot ../rockbot-game/
mv ./build/fonts ../rockbot-game/
mv ./build/shared ../rockbot-game/
mv ./build/games ../rockbot-game/
mv ./build/template ../rockbot-game/
mv ./build/strings_ingame.dat ../rockbot-game/
cd ../rockbot-game
```

Trying Rockbot 1. the game can't exit, i must force using : 
`killall -9 rockbot`

Trying Rockbot 2. Here the output
```
./rockbot
### PLAYER::CREATE::number[0]
get_filepath - FILEPATH:/media/Compartido/Liñux/cdf/git/rockbot-game/
main - argvString: './rockbot', FILEPATH: '/media/Compartido/Liñux/cdf/git/rockbot-game/'
SAVEPATH: /home/inukaze/.rockbot/
WARNING: Could not read config file '/home/inukaze/.rockbot/config_v204.sav'.
>> WII.DEBUG.INIT_GRAPHICS #1 <<
INFO: There are 0 joysticks attached
0 joysticks were found.

The names of the joysticks are:
>> WII.DEBUG.INIT_GRAPHICS #2 <<
>> WII.DEBUG.INIT_GRAPHICS #3 <<
>> WII.DEBUG.INIT_GRAPHICS.END <<
>>>>>>>>>>>>>>>>>>>> GAMENAME[Rockbot2]
>>> graphicsLib::load_icons <<<
>>> graphicsLib::load_icons #1 <<<
>>> graphicsLib::load_icons #1.1 <<<
>>> graphicsLib::load_icons #1.2[0] <<<
>>> graphicsLib::load_icons #1.3 <<<
>>> graphicsLib::load_icons #1.4 <<<
>>> graphicsLib::load_icons #1.5 <<<
>>> graphicsLib::load_icons #1.2[1] <<<
>>> graphicsLib::load_icons #1.3 <<<
>>> graphicsLib::load_icons #1.4 <<<
>>> graphicsLib::load_icons #1.5 <<<
>>> graphicsLib::load_icons #1.2[2] <<<
>>> graphicsLib::load_icons #1.3 <<<
>>> graphicsLib::load_icons #1.4 <<<
>>> graphicsLib::load_icons #1.5 <<<
>>> graphicsLib::load_icons #1.2[3] <<<
>>> graphicsLib::load_icons #1.3 <<<
>>> graphicsLib::load_icons #1.4 <<<
>>> graphicsLib::load_icons #1.5 <<<
>>> graphicsLib::load_icons #1.2[4] <<<
>>> graphicsLib::load_icons #1.3 <<<
>>> graphicsLib::load_icons #1.4 <<<
>>> graphicsLib::load_icons #1.5 <<<
>>> graphicsLib::load_icons #1.2[5] <<<
>>> graphicsLib::load_icons #1.3 <<<
>>> graphicsLib::load_icons #1.4 <<<
>>> graphicsLib::load_icons #1.5 <<<
>>> graphicsLib::load_icons #1.2[6] <<<
>>> graphicsLib::load_icons #1.3 <<<
>>> graphicsLib::load_icons #1.4 <<<
>>> graphicsLib::load_icons #1.5 <<<
>>> graphicsLib::load_icons #1.2[7] <<<
>>> graphicsLib::load_icons #1.3 <<<
>>> graphicsLib::load_icons #1.4 <<<
>>> graphicsLib::load_icons #1.5 <<<
>>> graphicsLib::load_icons #1.2[8] <<<
>>> graphicsLib::load_icons #1.3 <<<
>>> graphicsLib::load_icons #1.4 <<<
>>> graphicsLib::load_icons #1.5 <<<
>>> graphicsLib::load_icons #1.2[9] <<<
>>> graphicsLib::load_icons #1.3 <<<
>>> graphicsLib::load_icons #1.4 <<<
>>> graphicsLib::load_icons #1.5 <<<
>>> graphicsLib::load_icons #1.2[10] <<<
>>> graphicsLib::load_icons #1.3 <<<
>>> graphicsLib::load_icons #1.4 <<<
>>> graphicsLib::load_icons #1.5 <<<
>>> graphicsLib::load_icons #1.2[11] <<<
>>> graphicsLib::load_icons #1.3 <<<
>>> graphicsLib::load_icons #1.4 <<<
>>> graphicsLib::load_icons #1.5 <<<
>>> graphicsLib::load_icons #1.2[12] <<<
>>> graphicsLib::load_icons #1.3 <<<
>>> graphicsLib::load_icons #1.4 <<<
>>> graphicsLib::load_icons #1.5 <<<
>>> graphicsLib::load_icons #1.2[13] <<<
>>> graphicsLib::load_icons #1.3 <<<
>>> graphicsLib::load_icons #1.4 <<<
>>> graphicsLib::load_icons #1.5 <<<
>>> graphicsLib::load_icons #1.2[14] <<<
>>> graphicsLib::load_icons #1.3 <<<
>>> graphicsLib::load_icons #1.4 <<<
>>> graphicsLib::load_icons #1.5 <<<
>>> graphicsLib::load_icons #1.2[15] <<<
>>> graphicsLib::load_icons #1.3 <<<
>>> graphicsLib::load_icons #1.4 <<<
>>> graphicsLib::load_icons #1.5 <<<
>>> graphicsLib::load_icons #1.2[16] <<<
>>> graphicsLib::load_icons #1.3 <<<
>>> graphicsLib::load_icons #1.4 <<<
>>> graphicsLib::load_icons #1.5 <<<
>>> graphicsLib::load_icons #1.2[17] <<<
>>> graphicsLib::load_icons #1.3 <<<
>>> graphicsLib::load_icons #1.4 <<<
>>> graphicsLib::load_icons #1.5 <<<
>>> graphicsLib::load_icons #2 <<<
DEBUG.SDLSurfaceFromFile - Error in graphicsLib::SDLSurfaceFromFile - file '/media/Compartido/Liñux/cdf/git/rockbot-game/games/Rockbot2/images/tilesets/anim/lab_lava.png' not found
DEBUG.SDLSurfaceFromFile - Error in graphicsLib::SDLSurfaceFromFile - file '/media/Compartido/Liñux/cdf/git/rockbot-game/games/Rockbot2/images/tilesets/anim/lab_lava.png' not found
DEBUG.SDLSurfaceFromFile - Error in graphicsLib::SDLSurfaceFromFile - file '/media/Compartido/Liñux/cdf/git/rockbot-game/games/Rockbot2/images/tilesets/anim/lab_lava.png' not found
DEBUG.SDLSurfaceFromFile - Error in graphicsLib::SDLSurfaceFromFile - file '/media/Compartido/Liñux/cdf/git/rockbot-game/games/Rockbot2/images/tilesets/anim/lab_lava.png' not found
surfaceFromFile - error loading file: '/media/Compartido/Liñux/cdf/git/rockbot-game//games/Rockbot2/images/tilesets/anim/lab_lava.png'
show_debug_msg - msg: /media/Compartido/Liñux/cdf/git/rockbot-game//games/Rockbot2/images/tilesets/anim/lab_lava.png
graphicsLib::surfaceFromFile - error loading SDL surface from file '/media/Compartido/Liñux/cdf/git/rockbot-game//games/Rockbot2/images/tilesets/anim/lab_lava.png'.
show_debug_msg - msg: EXIT #05
terminate called after throwing an instance of 'std::invalid_argument'
  what():  Exception: file not found[graphicsLib::surfaceFromFile] - file[/media/Compartido/Liñux/cdf/git/rockbot-game//games/Rockbot2/images/tilesets/anim/lab_lava.png]
Backtrace:
[]
Abortado
```

i don't hear any sound. ok i try Rockbot 1 using the follow command line : 
`SDL_AUDIODRIVER=alsa ./rockbot`

Ready i hear music & sfx

Dunnos : 
1 - The Game Don't have any menu ?
2 - I can't change the video resolution ?
3 - I can't customize the controls ?

**Comment by inukaze** [comment]
Where i can download the files for Rockbot 2 : 

rockbot/build/games/Rockbot2/images/tilesets/anim/gear_001.png
rockbot/build/games/Rockbot2/images/tilesets/anim/gear_002.png
rockbot/build/games/Rockbot2/images/tilesets/anim/gear_003.png
rockbot/build/games/Rockbot2/images/tilesets/anim/gear_004.png
rockbot/build/games/Rockbot2/images/tilesets/anim/lab_lava.png
rockbot/build/games/Rockbot2/images/tilesets/anim/lab_lava_duct1.png
rockbot/build/games/Rockbot2/images/tilesets/anim/lab_lava_duct2.png

?

To make Rockbot 2 game works.

**Comment by protoman** [comment]
OK, I do believe the eror in Rockbot2 is tue to the accent in the path in
the "Liñux" directory.

To leave the game on linux, just press ESC key and a dialog should appear.

You can customize controls in the config menu.

Resolution is fixed in order to improve performance and due to being
multiple-platform. But you can change the code to use scale - there is a
code for 2x, you can change it in config also. Fell free to change
grapicslib code to use different resolutions.

Also, you can build the editor using Qt, most of the game can be changed
using it.
--
Iuri Fiedoruk, Software Developer

**Comment by protoman** [comment]
I've added the files to git. Sorry, I missed adding them.
--
Iuri Fiedoruk, Software Developer

**Comment by inukaze** [comment]
The game don't have any error with the spanish character "ñ" works perfectly. i thinks is thanks to the UTF-8. under GNU/Linux this avoid problems like that you can get on ANSI. 

Thanks for the files, Rockbot 2 now working fine. 

Last dunno -> How i can start my own spanish translation of game?

NOTE : When i press ESC and Select Exit and press "ENTER" key, the game don't exit, just hang up.

I should press ENTER to enter in the Weapon-Select Menu and press "W" key to get Engine-Menu Options. i think this engine-Menu should be on ESC or F1. and go to the "Quit Option" and press "X" because if i press "ENTER" the game Hang-Up

i notice i must press "X" key to quit the game properly, because if i press "ENTER" when i try to "Quit Game" the game Hang-Up

**Comment by protoman** [comment]
In build/shared, there is a strings_ingame_es.dat.
You will also need to enable the selection in the code. I think you just
need to remove the true parameter in the line:

options.push_back(st_menu_option("SPANISH", true));

On game_menu.cpp.


Open that in a file editor and change, but please be aware you need to keep
the number of characters more or less the same, otherwise they will be cut
by the dialog window.
--
Iuri Fiedoruk, Software Developer

**Comment by inukaze** [comment]
Ok don't worry the limitation of characters is not problem for people like me with some practice on romhacking of games of Nes and Sega Master System. :=)

**Comment by protoman** [comment]
Cool :)
BTW, you can send me your changes and I can merge them. Feel free to do
some crazy stuff, eheehhe.
--
Iuri Fiedoruk, Software Developer

**Comment by inukaze** [comment]
Hi there i had some things i want to say, because i feel you need fix it
Editing the file from translation i had notice of things like

When you press ENTER / RETURN you can open the menu in the right superior corner says
"CONFIG (R)". How i can change that "R" because for change configuration i must press "W" not "R"

The line 18 content "VIDEO" is not applied on game just appear 
"PC" on the Menu

The line 32 content "SET" is not necessary on Spanish. but leave a blank space if i delete SET. because on Spanish can be long word like "ASIGNAR" or "ESTABLECER"
and i dont know why the "set buttons" are don't transtale just keep : return, up, down, left, right

The line 106 content "MODE" is applied to
Menu -> Audio -> MODE : ON. This should be say "MUSIC"

Inside the Option "PC" here again the
MODE : Windowed / FullScreen

In the menu exist the option "PERFORMANCE" inside again the "MODE" : Low / Medium / High. and i don't idea whats mean. on spanish the common translation for PERFORMANCE are "Rendimiento" or "Actuacion" (this last refer to actors making their job meanwhile record a movie scene or something like that).

Dunnos : 
0 - This engine is just OpenSource or its FreeSoftware (Free of Freedom not from costless i prefer the term "Software Libre" because its more specific for this cases. )

1 - Why don't make a better structure file for translatations, like in the AM2R and their ini file (I make a spanish translation of game) how can check [**HERE**](https://github.com/inukaze/maestro/blob/master/Videojuegos/Traducciones/AM2R/V1.1/lang/spanish.ini)

Because [Some text are not translated] i need translate values like : 
(R)
RETURN
LEFT
RIGHT
UP
DOWN
FPS
MUSIC PLAYER : Inside the extra menu and inside music player
SFX PLAYER : Inside the extra menu and inside sfx player
SELECT GAME
(A) PLAY
(B) LEAVE
SELECT : On the Stage Select Screen
Military Base : On the Stage Select Screen
Power Station : On the Stage Select Screen
Circus : On the Stage Select Screen
Haunted Mansion : On the Stage Select Screen
Greek Ruins : On the Stage Select Screen
Ice Factory

Why i "leave stage" the first i enter in the center of level on stage screen select. i say messages : 
START-STG[11] DLG #0
START-STG[11] DLG #1
START-STG[11] DLG #2
START-STG[11] DLG P[1] 0
START-STG[11] DLG P[1] 1
START-STG[11] DLG P[1] 2
START-STG[11] DLG #3
START-STG[11] DLG #4
START-STG[11] DLG #5 
START-STG[11] DLG P[1] 3
START-STG[11] DLG P[1] 4
START-STG[11] DLG P[1] 5

I just think the first 3 are a dialog of protagonist and the next 3 are dialog with antagonist, the next 3 of protagonist and the last 3 of the antagonist

2 - Why for Music Volume and Sound Volume just exist the volume inside a sub-menu, instead of be in the menu and just adjust values using Left or Right arrows for example. because right now inside the sub-menu generate strange characters and the sub-menu i think are unnecessary 

In the test of Music Player and Sfx Player need be fixed i say something like
< 000?#>]? INTRO STAGE : On Music Player
< 000ø>]?

3 - Why in Rockbot 2, at the start if i go the left i fall, if i go to the left i found a ladder if i try to pass. the game hang up or just close with segmentation core [don't matter if i play the level normally at that point in the ladder happend the same]

4 - Why is the best way of i upload the modify version of your files to you using git ? just tell me the process with command lines :D 

5 - Stages on Rockbot2 :
Zodiac Level need the file -> /games/Rockbot2/images/sprites/objects/door_3.png 
Valkyrie Level need the file -> /games/Rockbot2/images/sprites/objects/door_4.png

For the moment i just make a partial translation -> [**HERE**](https://github.com/inukaze/maestro/tree/master/Videojuegos/Traducciones/Rockbot) <- because i don't know where i can see the translation applied. i just apply translation for the text i can watch.

**Comment by protoman** [comment]
Answers inline.
--
Iuri Fiedoruk, Software Developer


On Wed, Apr 24, 2019 at 4:54 AM Inukaze <notifications@github.com> wrote:

> Hi there i had some things i want to say, because i feel you need fix it
> Editing the file from translation i had notice of things like
>
> When you press ENTER / RETURN you can open the menu in the right superior
> corner says
> "CONFIG (R)". How i can change that "R" because for change configuration i
> must press "W" not "R"
>
Yeah, this refers to the button or key mapped as "R", not the actual key.

> The line 18 content "VIDEO" is not applied on game just appear
> "PC" on the Menu
>
There are some strings that were later removed. I kept them, as it would
mess with the enum.

> The line 32 content "SET" is not necessary on Spanish. but leave a blank
> space if i delete SET. because on Spanish can be long word like "ASIGNAR"
> or "ESTABLECER"
> and i dont know why the "set buttons" are don't transtale just keep :
> return, up, down, left, right
>
You may have reached a part of the code that does not use i18n. I've added
it later, so I may have missed some spots.

> The line 106 content "MODE" is applied to
> Menu -> Audio -> MODE : ON. This should be say "MUSIC"
>
Actually, it means that both music and sound effects are enabled or not.
May not be the best way to phrase it out, but is not wrong.

> Inside the Option "PC" here again the
> MODE : Windowed / FullScreen
>
Yes, it means the way the window should be presented, thus, mode.

> In the menu exist the option "PERFORMANCE" inside again the "MODE" : Low /
> Medium / High. and i don't idea whats mean. on spanish the common
> translation for PERFORMANCE are "Rendimiento" or "Actuacion" (this last
> refer to actors making their job meanwhile record a movie scene or
> something like that).
>
Well, in portuguese, performance translates directly as "performance" :)
This is a mode that you can set a few special effects to be turned on or
off, depending on your machine's performance. If it runs the game slow, set
to low and it will disable all special effects and also remove the parallax
effect on background. In high, it enables all special effects such as
transparencies in a few stages.

> Dunnos :
> 1 - Why don't make a better structure file for translatations, like in the
> AM2R and their ini file (I make a spanish translation of game) how can
> check *HERE*
> <https://github.com/inukaze/maestro/blob/master/Videojuegos/Traducciones/AM2R/V1.1/lang/spanish.ini>
>
KISS
This was added later nto the game, as I said, and used only by me until now
:)

> Because [Some text are not translated] i need translate values like :
> (R)
> RETURN
> LEFT
> RIGHT
> UP
> DOWN
> FPS
> MUSIC PLAYER : Inside the extra menu and inside music player
> SFX PLAYER : Inside the extra menu and inside sfx player
> SELECT GAME
> (A) PLAY
> (B) LEAVE
> SELECT : On the Stage Select Screen
> Military Base : On the Stage Select Screen
> Power Station : On the Stage Select Screen
> Circus : On the Stage Select Screen
> Haunted Mansion : On the Stage Select Screen
> Greek Ruins : On the Stage Select Screen
> Ice Factory
>
MUSIC PLAYER, SELECT GAME and others are stored in
build/shared/strings_ingame_es.dat.

About other strings, some are in dialogs:
In games/Rockbot1/dialogs there are the messages that appear when you start
a stage or reach the boss room.
Other, such as stage names, can't be translated now, I've added no support
for it.

Why i "leave stage" the first i enter in the center of level on stage
> screen select. i say messages :
> START-STG[11] DLG #0
> START-STG[11] DLG #1 <https://github.com/protoman/rockbot/issues/1>
> START-STG[11] DLG #2 <https://github.com/protoman/rockbot/issues/2>
> START-STG[11] DLG P[1] 0
> START-STG[11] DLG P[1] 1
> START-STG[11] DLG P[1] 2
> START-STG[11] DLG #3 <https://github.com/protoman/rockbot/issues/3>
> START-STG[11] DLG #4 <https://github.com/protoman/rockbot/issues/4>
> START-STG[11] DLG #5 <https://github.com/protoman/rockbot/issues/5>
> START-STG[11] DLG P[1] 3
> START-STG[11] DLG P[1] 4
> START-STG[11] DLG P[1] 5
>
Those are just debug messages I've added into the code.

> I just think the first 3 are a dialog of protagonist and the next 3 are
> dialog with antagonist, the next 3 of protagonist and the last 3 of the
> antagonist
>
> 2 - Why for Music Volume and Sound Volume just exist the volume inside a
> sub-menu, instead of be in the menu and just adjust values using Left or
> Right arrows for example. because right now inside the sub-menu generate
> strange characters and the sub-menu i think are unnecessary
>
Feel free to fix or change the way settings are defined :)

> In the test of Music Player and Sfx Player need be fixed i say something
> like
> < 000?#>]? INTRO STAGE : On Music Player
> < 000ø>]?
>
This can happen when you add a new line in translations. Make sure the
number of lines in your translatted file is the same as in the original
(english) one.

> 3 - Why in Rockbot 2, at the start if i go the left i fall, if i go to the
> left i found a ladder if i try to pass. the game hang up or just close with
> segmentation core [don't matter if i play the level normally at that point
> in the ladder happend the same]
>
That is a debug left - rememver you are building the dev tree ;)

> 4 - Why is the best way of i upload the modify version of your files to
> you using git ? just tell me the process with command lines :D
>
Oh, I am not sure, but I know you can create a copy of the project into
your github account and then ask for a merge into my code. But I've used
this only once, so I don't recall the commands.

**Comment by inukaze** [comment]
> MUSIC PLAYER, SELECT GAME and others are stored in build/shared/strings_ingame_es.dat.

Are not translated. if you try with my [**Partial translation to spanish**](https://github.com/inukaze/maestro/blob/master/Videojuegos/Traducciones/Rockbot/build/shared/strings_ingame_es.dat)

> Feel free to fix or change the way settings are defined :)
i don't know how i can delete the sub menu and just leave
MUSIC VOLUME         :       128
SFX      VOLUME        :       096

if you question why the long blank space, its for translations because in spanish should be something like
VOLUMEN DE MUSICA    : 128
VOLUMEN DE EFECTOS  : 096
Mockup -> [![Rockbot-EN-Audio-Menu-Mockup.png](https://i.postimg.cc/VsgRV5wn/Rockbot-EN-Audio-Menu-Mockup.png)](https://postimg.cc/V5SnSfdk)

and on english looks equal because this section its not translated. with english text : 
MUSIC VOLUME -> [![Rockbot-EN-Music-Volume.png](https://i.postimg.cc/Nj5pTtWT/Rockbot-EN-Music-Volume.png)](https://postimg.cc/MnJ72CZK)

SFX VOLUME -> [![Rockbot-EN-SFXVolume.png](https://i.postimg.cc/gkMgnjBm/Rockbot-EN-SFXVolume.png)](https://postimg.cc/hfdLYDB5)

**Comment by protoman** [comment]
There is something odd here.
Look at my screen:
[image: image.png]

Notice that it does not show the volume inline. I removed that a while ago.
Where did you pick the code from, git?
--
Iuri Fiedoruk, Software Developer

**Comment by inukaze** [comment]
> There is something odd here. Look at my screen: [image: image.png] Notice that it does not show the volume inline. I removed that a while ago. Where did you pick the code from, git? -- Iuri Fiedoruk, Software Developer

Notice its a manual image made i called it "Mockup"

The code i had compile its from this github, just observe the top post.  is the follow :

```
git clone https://github.com/protoman/rockbot
cd rockbot
./fetch.sh
sed -i '8d' ./Rockbot.pro
sed -i '17d' ./Rockbot.pro
sed -i '13 s/^.//' ./Rockbot.pro
qmake
make
```

**Comment by hardcaz** [comment]
how do i compile if i am using windows 10

**Comment by protoman** [comment]
@cypherpunks132 hey. I will suppose you want to build the windows version. First you need to get the Qt libraries (to build the editor and because it is used in build system) and SDL. Make sure to get the mingw for both. I recommend getting qt creator, the IDE I do use, it makes easier to avoid having to use command line.

You need to install qmake (from qt library) to generate the Makefile for your Windows 10. There is a file in the main folder, with .pro exptesion, open it, comment out line CONFIG += linux, and uncomment CONFIG += win32.
Go to the win32 block and adjust the folders (ignore the ones like /usr/lib, those are for mingw, leave as is).

Run qmake, eg "qmake.exe RockDroid.pro". It will generate a Makefile, now simply run make and it should build. Please contact me by email if you need further instructions (protoman@upperland.net).

BTW, there are two branches, master that is the old-style game, and rockdroid that is the current game.

**Comment by hardcaz** [comment]
@protoman thanks so what options do you recommend i select with the qt online installer also do i use sdl 1.2 or sdl 2.0 i use gmailnator so i cant send you an email so i used emkei's fake mailer

**Comment by protoman** [comment]
Sorry, I do not recall what are the options in the installer, so I can't
give you that :)
--
Iuri Fiedoruk, Software Developer

**Comment by hardcaz** [comment]
@protoman where do i place the qmake.exe and sdl files

**Comment by hardcaz** [comment]
@protoman i think i am gonna further instructions

**Comment by protoman** [comment]
No problem, building first time is no easy task.
Sorry I can't reply as fast as possible, I have a daily job, because games
do not give me any money :)
Let me check my windows VM to see the directories and such.
--
Iuri Fiedoruk, Software Developer

**Comment by hardcaz** [comment]
@protoman ok just tell me when you have checked your windows vm

**Comment by protoman** [comment]
OK, I've installed Qt in C:\Qt\4.8.4\lib (as you can see, this is an older
version of qt, but you can just adjust the version).
Inside it, there is a lib/x86 folder that contains the SDL libraries needed
for building. Here is the folder contents:

20/01/2012  23:01           328.192 libFLAC-8.dll
20/01/2012  23:03           461.824 libfreetype-6.dll
20/01/2012  22:54           204.288 libjpeg-8.dll
20/01/2012  23:01           263.168 libmikmod-2.dll
20/01/2012  23:01            24.064 libogg-0.dll
20/01/2012  22:54           151.552 libpng15-15.dll
06/04/2009  01:13            71.102 libSDL_gfx.a
06/04/2009  01:13             3.182 libSDL_gfx.def
20/01/2012  22:54           408.064 libtiff-5.dll
20/01/2012  23:01           163.840 libvorbis-0.dll
20/01/2012  23:01            36.352 libvorbisfile-3.dll
20/01/2012  22:54           180.224 libwebp-2.dll
20/01/2012  23:01             1.649 LICENSE.FLAC.txt
20/01/2012  23:03             7.033 LICENSE.freetype.txt
20/01/2012  22:54             2.998 LICENSE.jpeg.txt
20/01/2012  23:01            25.892 LICENSE.mikmod.txt
20/01/2012  23:01             1.625 LICENSE.ogg-vorbis.txt
20/01/2012  22:54             4.438 LICENSE.png.txt
20/01/2012  23:01            25.873 LICENSE.smpeg.txt
20/01/2012  22:54             1.294 LICENSE.tiff.txt
20/01/2012  22:54             1.653 LICENSE.webp.txt
20/01/2012  23:03             1.446 LICENSE.zlib.txt
06/04/2009  01:13            62.464 SDL_gfx.dll
20/01/2012  02:00            51.200 SDL_image.dll
05/01/2012  21:43             8.650 SDL_image.lib
15/01/2012  13:50           160.256 SDL_mixer.dll
05/01/2012  21:46            16.784 SDL_mixer.lib
15/01/2012  13:50            27.136 SDL_ttf.dll
05/01/2012  22:53            11.168 SDL_ttf.lib
20/01/2012  23:01           209.408 smpeg.dll
20/01/2012  23:03           100.352 zlib1.dll

Sabe for the lib/x64 folder, but with the 64 bits version of the libraries:
20/01/2012  23:01           306.176 libFLAC-8.dll
20/01/2012  23:03           484.352 libfreetype-6.dll
20/01/2012  22:54           200.192 libjpeg-8.dll
20/01/2012  23:01           252.416 libmikmod-2.dll
20/01/2012  23:01            25.600 libogg-0.dll
20/01/2012  22:54           211.456 libpng15-15.dll
20/01/2012  22:54           466.944 libtiff-5.dll
20/01/2012  23:01           187.392 libvorbis-0.dll
20/01/2012  23:01            36.352 libvorbisfile-3.dll
20/01/2012  22:54           222.208 libwebp-2.dll
20/01/2012  23:01             1.649 LICENSE.FLAC.txt
20/01/2012  23:03             7.033 LICENSE.freetype.txt
20/01/2012  22:54             2.998 LICENSE.jpeg.txt
20/01/2012  23:01            25.892 LICENSE.mikmod.txt
20/01/2012  23:01             1.625 LICENSE.ogg-vorbis.txt
20/01/2012  22:54             4.438 LICENSE.png.txt
20/01/2012  23:01            25.873 LICENSE.smpeg.txt
20/01/2012  22:54             1.294 LICENSE.tiff.txt
20/01/2012  22:54             1.653 LICENSE.webp.txt
20/01/2012  23:03             1.446 LICENSE.zlib.txt
20/01/2012  02:00            53.248 SDL_image.dll
09/01/2012  07:08             8.478 SDL_image.lib
15/01/2012  13:50           161.792 SDL_mixer.dll
09/01/2012  07:18            16.424 SDL_mixer.lib
15/01/2012  13:50            29.184 SDL_ttf.dll
09/01/2012  07:23            10.970 SDL_ttf.lib
20/01/2012  23:01           227.328 smpeg.dll
20/01/2012  23:03            90.624 zlib1.dll

--
Iuri Fiedoruk, Software Developer

**Comment by protoman** [comment]
If you want, I can create a ZIP with those two folders (x86 and x64). But
you still need to install Qt and mingw in your computer.
--
Iuri Fiedoruk, Software Developer

**Comment by protoman** [comment]
Here it goes:
https://drive.google.com/file/d/1TTMfiUjIGsydOCkLd_32D56PGyRNkdEA/view?usp=sharing
--
Iuri Fiedoruk, Software Developer

**Comment by hardcaz** [comment]
@protoman i extracted the lib.7z file in c:/qt/5.15.0/mingw81_32 and c:/qt/5.15.0/mingw81_64 what do i do now

**Comment by protoman** [comment]
OK, I don't know if those folders (mingw81_32 and mingw81_64) will work,
but let's go ahead.
Did you install qt creator?
If so, after cloning the code from github, just open the project
(Rockbot.pro or Rockdroid.pro) in Creator and try to build.
--
Iuri Fiedoruk, Software Developer

**Comment by hardcaz** [comment]
@protoman didnt work i got the error that there was no file with the name sdl.h

**Comment by protoman** [comment]
Yes, I think I forgot I need to send you the include directory also 🤣

Please give me a while, I am in a meeting for deploying a new software on
my job.

Em ter, 7 de jul de 2020 14:38, cypherpunks132 <notifications@github.com>
escreveu:

**Comment by protoman** [comment]
https://drive.google.com/file/d/12nUXHaXR7CjHPIEI58KWpgv3zvr6KtE3/view?usp=sharing

Unzip this in the qt/include folder.
--
Iuri Fiedoruk, Software Developer

**Comment by hardcaz** [comment]
@protoman you forgot to set it to public

**Comment by protoman** [comment]
The link should make public to anyone downloadig it.

Em ter, 7 de jul de 2020 15:07, cypherpunks132 <notifications@github.com>
escreveu:

**Comment by hardcaz** [comment]
no i says for me that i need permission

**Comment by protoman** [comment]
So ask for it and I will receive tue request and allow.

Em ter, 7 de jul de 2020 15:14, cypherpunks132 <notifications@github.com>
escreveu:

**Comment by hardcaz** [comment]
it didnt work i got the error couldn't send request edit: fixed

**Comment by protoman** [comment]
I granted the permission.
--
Iuri Fiedoruk, Software Developer

**Comment by hardcaz** [comment]
it errors out again it says cannot find -lSDL

**Comment by protoman** [comment]
That is great. Now check the .pro file for the link parameters. Ensure it
points to the libraries folder you have in your system.

Em ter, 7 de jul de 2020 15:54, cypherpunks132 <notifications@github.com>
escreveu:

**Comment by protoman** [comment]
LIBS =  -lSDL_mixer -lSDL_image -lSDL_ttf -lSDL_gfx -lmingw32 -lSDL
-mwindows -LC:\Qt\4.8.4\lib -L/usr/lib -L/usr/lib/x86 -LC:\MinGW\lib
-LC:\MinGW\lib\x86


Make sure this line points to the folder you have the x86 and x64
folders - probably you need to use those names, and not mingw81*.

--
Iuri Fiedoruk, Software Developer

**Comment by hardcaz** [comment]
what do i do with the -l\usr\lib in the .pro

**Comment by protoman** [comment]
Check my previous instructions...

Em ter, 7 de jul de 2020 15:59, cypherpunks132 <notifications@github.com>
escreveu:

**Comment by hardcaz** [comment]
yeah i think i might need more infomation

**Comment by protoman** [comment]
All the information you need I've already wrote. Now just research on
google about linking errors. Bit it is jist a matter of pointing to the
riggr folder and having the right names for the x86 and x64 libs.

Em ter, 7 de jul de 2020 16:14, cypherpunks132 <notifications@github.com>
escreveu:

**Comment by hardcaz** [comment]
i think i am gonna install mingw with gcc 4.4 and qt 4.8.4 to make sure everything works first try

**Comment by protoman** [comment]
There is no need to use same version, just adjust the folders correctly.
BTW, why do you want to build the game, to create a custom?
--
Iuri Fiedoruk, Software Developer

**Comment by hardcaz** [comment]
so i can play rockdroid on pc and play the dev version of rockbot (rockdroid before it get taken down from google play)

**Comment by protoman** [comment]
You could just asked me. I did not made a version because I was having
issues with the editor to build. I think I need to create a windows 7 VM
for building, my win XP vm is not good for newer versions of Qt and such.
But it is actually great you do all that trouble to play it :)
--
Iuri Fiedoruk, Software Developer

**Comment by hardcaz** [comment]
yeah but i dont know what folders i need to change in the .pro file i could send the .pro file to you so you can help me adjust the folders

**Comment by protoman** [comment]
-LC:\Qt\4.8.4\lib

--
Iuri Fiedoruk, Software Developer

**Comment by protoman** [comment]
That is the folder you need to change to your existing qt install.
Also, I don't know, but maybe you need to rename mingw81* to just x86 and
x65.
--
Iuri Fiedoruk, Software Developer

**Comment by hardcaz** [comment]
@protoman you mean x64 right

**Comment by hardcaz** [comment]
so i need to change their names from mingw81_64 to mingwx64 and mingw_32 to mingwx32

**Comment by hardcaz** [comment]
also what do i do with the makefile

**Comment by hardcaz** [comment]
also i did rename -LC:\Qt\4.8.4\lib to -LC:\Qt\5.15.0\lib

**Comment by hardcaz** [comment]
and i still get the error cannot find -lSDL

**Comment by hardcaz** [comment]
and the error is C:/Qt/Tools/mingw810_32/bin/../lib/gcc/i686-w64-mingw32/8.1.0/../../../../i686-w64-mingw32/bin/ld.exe: cannot find -lSDL

**Comment by hardcaz** [comment]
i cant see the full error

**Comment by hardcaz** [comment]
and there is another error which is collect2.exe: error: ld returned 1 exit status

**Comment by protoman** [comment]
Something is wrong in your path and it is not finding libsdl.a, this is the
issue, you need to fix your path, don't rename them on your windows, just
point to the right path in Rockbot.pro, if needed, add entries for your lib
folders where you placed the SDL library files.

The last error is just linker telling you it has failed.

Please be patient (or use google) while I am building a win7 vm with more
recent versions of the SDK, so I can give you better instructions, if
possible,
--
Iuri Fiedoruk, Software Developer

**Comment by hardcaz** [comment]
i cant find the file libsdl.a

**Comment by protoman** [comment]
Yeah, I missed that. Please wait.
--
Iuri Fiedoruk, Software Developer

**Comment by hardcaz** [comment]
ok

**Comment by protoman** [comment]
https://drive.google.com/file/d/1EgUtn6PAY3Dbqa_nNEsaVAjZIIwXsMoQ/view?usp=sharing
Put those in your qt/mingw/lib folder.

https://drive.google.com/file/d/1J_C9ybinaq6NqwYXxolsFSD5J65EddxR/view?usp=sharing
Store that for later, when running the game.

The configuration I used:
Version:1.0 StartHTML:0000000107 EndHTML:0000003931
StartFragment:0000000127 EndFragment:0000003893


win32 {

    DEFINES += WIN32

    LIBS =  -lSDL_mixer -lSDL_image -lSDL_ttf -lSDL_gfx -lmingw32
-lSDL -mwindows -LC:\Qt\5.15.0\mingw81_32\lib -L/usr/lib
-L/usr/lib/x86 -LC:\MinGW\lib -LC:\MinGW\lib\x86
-LC:\Qt\5.15.0\mingw81_32\lib\x86

    INCLUDES = -I/usr/include/SDL -I/usr/include -I. -I./include
-L/usr/lib -IC:\Qt\5.15.0\mingw81_32\include

    QMAKE_CCFLAGS += -DWIN32 -DPC

    QMAKE_CXXFLAGS += -DWIN32 -DPC -IC:\Qt\5.15.0\mingw81_32\
-IC:\Qt\5.15.0\mingw81_32\include -LC:\Qt\5.15.0\mingw81_32\lib

    CONFIG -= console

}


With that I was able to build.

For running it, there are a few extra steps needed, but let's first
try to compile it ;)

--
Iuri Fiedoruk, Software Developer

**Comment by protoman** [comment]
Ignore this line (don't know how it came with a copy/paste):
Version:1.0 StartHTML:0000000107 EndHTML:0000003931
StartFragment:0000000127 EndFragment:0000003893
--
Iuri Fiedoruk, Software Developer

**Comment by hardcaz** [comment]
error: [Makefile:245: Makefile] Error 3

**Comment by protoman** [comment]
I can't help you with Makefile errors, sorry.
OK, first, send me a direct email to protoman@upperland.net, let's stop
doing that through github. And send me your .pro file attached.
--
Iuri Fiedoruk, Software Developer

**Comment by hardcaz** [comment]
done

**Comment by hardcaz** [comment]
when i try to compile the master branch it starts but it starts directly in a stage instead of the main menu

**Comment by protoman** [comment]
This is just because quickload is true. It is a debug option so I can test
with less time until getting im the stage.
I will make a new build later and remove it.

Em qua, 8 de jul de 2020 13:03, cypherpunks132 <notifications@github.com>
escreveu:

**Comment by hardcaz** [comment]
how do i set quickload to false

**Comment by protoman** [comment]
On main.cpp, check for GAME_FLAGS[FLAG_QUICKLOAD] = true;
Comment out and rebuild.
--

Iuri Fiedoruk, Software Developer

**Comment by hardcaz** [comment]
so change true to false

**Comment by protoman** [comment]
On main.cpp, check for GAME_FLAGS[FLAG_QUICKLOAD] = true;
*Comment out* and rebuild.
--
Iuri Fiedoruk, Software Developer

**Comment by hardcaz** [comment]
or do i remove the # from #itdef pc

**Comment by protoman** [comment]
Man, I have to work. Google it.
If you don't know how to comment out a line of code in C++, I am not the
one who will teach you.

Em qua, 8 de jul de 2020 14:02, cypherpunks132 <notifications@github.com>
escreveu:

**Comment by hardcaz** [comment]
i did comment out but nothing changed

**Comment by hardcaz** [comment]
i am not sure what point in the line i am supposed to comment out

**Comment by hardcaz** [comment]
also there are multiple GAME_FLAGS[FLAG_QUICKLOAD] = true;

**Comment by hardcaz** [comment]
fixed

**Comment by hardcaz** [comment]
can anyone help me compile the old commit 1182e7dbfe3b9ee2174ea0b4218dbb9747c89cd2

**Comment by humbertodias** [comment]
@inukaze try out https://github.com/protoman/rockbot/pull/144

**Comment by inukaze** [comment]
> [@inukaze](https://github.com/inukaze) try out [#144](https://github.com/protoman/rockbot/pull/144)

root[tmp]# `git clone https://github.com/protoman/rockbot`

> Clonando en 'rockbot'...
> remote: Enumerating objects: 18206, done.
> remote: Counting objects: 100% (139/139), done.
> remote: Compressing objects: 100% (67/67), done.
> remote: Total 18206 (delta 92), reused 72 (delta 72), pack-reused 18067 (from 3)
> Recibiendo objetos: 100% (18206/18206), 305.43 MiB | 10.94 MiB/s, listo.
> Resolviendo deltas: 100% (12564/12564), listo.
> Actualizando archivos: 100% (2032/2032), listo.

root[tmp]# `cd rockbot/`
root[rockbot]# `./fetch.sh` 
Ya está actualizado.
root[rockbot]# `qmake Rockbot.pro `

> WARNING: Failure to find: docs/game_manual.cpp


root[rockbot]# `make`

> /bin/bin2s ../ports/ps2/SJPCM.irx ../ports/ps2/SJPCM.s SJPCM
> make: /bin/bin2s: No existe el fichero o el directorio
> make: *** [Makefile:259: SJPCM.s] Error 127

**Comment by humbertodias** [comment]
@inukaze

the main branch is outdated and the [PR#145](https://github.com/protoman/rockbot/pull/145) is still pending 
So, try out
```shell
git clone -b sdl2 https://github.com/humbertodias/rockbot
cd rockbot
./build-ubuntu.sh
cd build && ./rockbot
``` 

<img width="964" height="744" alt="Image" src="https://github.com/user-attachments/assets/5d7fe8e0-0670-4bfb-9814-3c2c04a77d1e" />

alternatively, you can download the compiled version from:
[https://github.com/humbertodias/rockbot/releases/tag/v2.01.065](https://github.com/humbertodias/rockbot/releases/tag/v2.01.065)

or try the web version here:
[https://humbertodias.github.io/rockbot/](https://humbertodias.github.io/rockbot/)

**Comment by inukaze** [comment]
> [@inukaze](https://github.com/inukaze)
> 
> the main branch is outdated and the [PR#145](https://github.com/protoman/rockbot/pull/145) is still pending So, try out
> 
> git clone -b sdl2 https://github.com/humbertodias/rockbot
> cd rockbot
> ./build-ubuntu.sh
> cd build && ./rockbot
> <img alt="Image" width="964" height="744" src="https://private-user-images.githubusercontent.com/9255997/513660661-5d7fe8e0-0670-4bfb-9814-3c2c04a77d1e.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjMwNzU0MjUsIm5iZiI6MTc2MzA3NTEyNSwicGF0aCI6Ii85MjU1OTk3LzUxMzY2MDY2MS01ZDdmZThlMC0wNjcwLTRiZmItOTgxNC0zYzJjMDRhNzdkMWUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MTExMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTExMTNUMjMwNTI1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9M2RiOWI1ZTg2Y2ZlZWM1YzZlNGQwM2Y4MjE5NjYzNzYwZDU1YzgyMTRmNjgzZGMwMjE1OWFlOWI2MjFhODBlNSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.E-Uq6P4doiYxmSYFUTkbfLeZ1SOdgFfDj9vwMQeLt4Q">
> alternatively, you can download the compiled version from: https://github.com/humbertodias/rockbot/releases/tag/v2.01.065
> 
> or try the web version here: https://humbertodias.github.io/rockbot/

root[tmp]# `git clone -b sdl2 https://github.com/humbertodias/rockbot`

> Clonando en 'rockbot'...
> remote: Enumerating objects: 18593, done.
> remote: Counting objects: 100% (146/146), done.
> remote: Compressing objects: 100% (75/75), done.
> remote: Total 18593 (delta 91), reused 83 (delta 71), pack-reused 18447 (from 3)
> Recibiendo objetos: 100% (18593/18593), 343.64 MiB | 10.93 MiB/s, listo.
> Resolviendo deltas: 100% (12830/12830), listo.
> root[tmp]# cd rockbot
> root[rockbot]# ./build-ubuntu.sh
> 🛠️ Updating packages and installing dependencies...
> sudo: apt: command not found
> sudo: apt: command not found
> 

root[rockbot]#` cd build && ./rockbot`

> bash: ./rockbot: No existe el fichero o el directorio

root[build]# 

<img width="1280" height="1024" alt="Image" src="https://github.com/user-attachments/assets/0dce9cbb-779b-4703-820c-c95fa15f430b" />

**Comment by humbertodias** [comment]
```shell
./build-ubuntu.sh
apt command not found
```
Looks like you’re not on Ubuntu. The script is intended for Ubuntu/Debian.

Feel free to create an additional script named `build-YOURDISTRO.sh`.

You can use this file as a reference:

[https://github.com/humbertodias/rockbot/blob/sdl2/build-ubuntu.sh](https://github.com/humbertodias/rockbot/blob/sdl2/build-ubuntu.sh)

In most cases, the only change you’ll need is to swap out the package manager - replace `apt` with your distribution’s tool (`yum`, `rpm`, `pacman`, etc.).

**Comment by inukaze** [comment]
> ./build-ubuntu.sh
> apt command not found
> Looks like you’re not on Ubuntu. The script is intended for Ubuntu/Debian.
> 
> Feel free to create an additional script named `build-YOURDISTRO.sh`.
> 
> You can use this file as a reference:
> 
> https://github.com/humbertodias/rockbot/blob/sdl2/build-ubuntu.sh
> 
> In most cases, the only change you’ll need is to swap out the package manager - replace `apt` with your distribution’s tool (`yum`, `rpm`, `pacman`, etc.).

i am writing right now, but is some hard for me, well i just try to make it one script for this purpose.

**Comment by humbertodias** [comment]
Which distribution are you using?

**Comment by inukaze** [comment]
> Which distribution are you using?
Slackware, but that is not relevant. 

i prefer make a macro-script and use things like : 
`lsb_release -d | awk -F'\t' '{print $2}' | cut -d' ' -f1-2` and compare with array values, and in a other variable put the list of files need the binary executable for create it . well i try wrote full the saturday.

a macro-script but very uncompleted because i am bad writing source code. and i not trust on ai for make it xD the ia i just can use like a guide for understand the lines i want put on the file XD

**Comment by humbertodias** [comment]
No problem. Try out
https://github.com/humbertodias/rockbot/blob/sdl2/build-slackware.sh

**Comment by inukaze** [comment]
This is my modification of that script becuase i use slapt-get : 
[build-slackware.sh](https://github.com/user-attachments/files/23537671/build-slackware.sh)

**Comment by humbertodias** [comment]
> This is my modification of that script becuase i use slapt-get : [build-slackware.sh](https://github.com/user-attachments/files/23537671/build-slackware.sh)

Awesome! Did it work properly?

**Comment by inukaze** [comment]
> > This is my modification of that script becuase i use slapt-get : [build-slackware.sh](https://github.com/user-attachments/files/23537671/build-slackware.sh)
> 
> Awesome! Did it work properly?

Yes, musicless but works

**Comment by humbertodias** [comment]
Great! 
The game uses `SDL_Mixer` that talks with the real implementation `ALSA`, `PulseAudio`, `PipeWire`, etc. As you can see
```
┌─────────────────────────┐
│       Application       │
│         rockbot         │
└───────────┬─────────────┘
            │
            ▼
┌─────────────────────────┐
│        SDL2             │
│   (libSDL2-2.0)         │
│ - Initializes audio     │
│ - Manages audio buffers │
└───────────┬─────────────┘
            │
            ▼
┌─────────────────────────────┐
│     SDL2_mixer              │
│  (libSDL2_mixer)            │
│ - Audio mixing              │
│ - Multi-channel support     │
└───────────┬─────────────────┘
            │
 ┌──────────┼───────────┐
 ▼          ▼           ▼
┌────────┐ ┌─────────┐ ┌─────────┐
│ WAV    │ │ MP3     │ │ OGG/Vorbis │
│ libsndfile │ libmpg123 │ libvorbisfile │
└────────┘ └─────────┘ └─────────┘
            │           │
            ▼           ▼
      ┌─────────────┐ ┌─────────┐
      │ FLAC        │ │ Opus    │
      │ libFLAC.so  │ │ libopus │
      └─────────────┘ └─────────┘
            │
            ▼
┌─────────────────────────┐
│ ALSA (libasound.so)     │
│ PulseAudio (libpulse.so)│
│ - Final audio output    │
└─────────────────────────┘
```
I recommend double-checking your local setup. It’s working here with `SDL2_mixer` using `ALSA`. 
Try that to identify
```bash
if pgrep -x pulseaudio > /dev/null; then
    echo "PulseAudio is running"
elif pgrep -x pipewire > /dev/null; then
    echo "PipeWire is running"
else
    echo "Only ALSA is available"
fi
```
Let us know the result. Thanks

**Comment by protoman** [comment]
Sorry for the delay. Life got into the in the last year, things were not
easy, but I'm trying to get back into dedicating some time to Rockbot.

**Comment by inukaze** [comment]
Slackware from 14.2 came with pulseaudio system by default.

Becuase i says on the forum LinuxQuestion of Slackware 14.1 why don't became with pulseaudio, i hate it, and the software of leonerd potering because i think its bad for GNU/Linux in general, but sadly is the most standard, but just adapt pulseaudio.

Because for reasons i dont know, dont understand, and not comprend, nobody never had adapted the ALSA source code to work with simultaneous programs audio output like pulseaudio, i watch a look on internet much new devs want destroy alsa, and that is a non-sense, alsa is part of kernel, pulseaudio not. pulseaudio require alsa to work, but alsa not requiered pulseaudio to work.

[ inukaze  | 14-11-2025 | 05:03 pm ]
[build]$` ./rockbot `

> WARNING: Could not read config file '/home/inukaze/.rockdroid/config_v205.sav'.
> Renderer backend: opengl
> SDL version: 2.0.20
> Renderer backend: opengl
> SDL version: 2.0.20
> Renderer backend: opengl
> SDL version: 2.0.20
> Error in soundLib::load_boss_music::Mix_LoadMUS('/tmp/rockbot/build/games/RockDroid1/music/rockbot_boss.mod': 'Unrecognized audio format'
> FIO:_STRINGS::get_string_list_from_scene_text_file[/tmp/rockbot/build//games/RockDroid1/scenes/text/en/0.txt]
> font-size[8], max_line_w[25], adjust_x[96], center_x[64]
> Error in soundLib::load_music::Mix_LoadMUS('/tmp/rockbot/build/games/RockDroid1/music/rockbot_opening.mod': 'Unrecognized audio format'
> ERROR: play_music, music is null
> RESET cheat input - expected-button[0], pressed-button[1]
> RESET cheat input - expected-button[0], pressed-button[1]
> RESET cheat input - expected-button[0], pressed-button[7]
> CONTINUE cheat input
> CONTINUE cheat input
> CONTINUE cheat input
> RESET cheat input - expected-button[1], pressed-button[0]
> RESET cheat input - expected-button[0], pressed-button[7]
> Error in soundLib::load_music::Mix_LoadMUS('/tmp/rockbot/build/games/RockDroid1/music/rockbot_chaos_city.mod': 'Unrecognized audio format'
> ERROR: play_music, music is null
> ### CHARACTER::REMOVE PLATFORM #2
> ### JUMP::START - bigjump_mode[0]
> ### CHARACTER::REMOVE PLATFORM #2
> ### JUMP::START - bigjump_mode[0]
> ### CHARACTER::REMOVE PLATFORM #2
> ### JUMP::START - bigjump_mode[0]
> 

Questions:

1 - Why is the music in "mod" format?

2 - Do you plan to make a version for MS-DOS 5.22?

3 - Does the music format only support mod, not "it" or "xm"?

4 - Will you make it use "mid" (midis) through Timidity++ or FluidSynth?

Although, you'll have to use a soundfont (SF2) from a video game like Mega Man for NES or Pulseman for Sega Genesis to better fit the atmosphere of your game?

Another things i notice, if i enter on the on music volume and return to menu
<img width="1280" height="1024" alt="Image" src="https://github.com/user-attachments/assets/98ec532c-21b9-4823-94c9-42c5c1611c37" />
<img width="1280" height="1024" alt="Image" src="https://github.com/user-attachments/assets/f2e942bb-fcc5-4b58-aa05-8451d7874e0d" />

The text on the first file looks weird
<img width="1280" height="1024" alt="Image" src="https://github.com/user-attachments/assets/58bcd178-9fa8-4bfd-a638-e01306ad3000" />

Ok some area lost the same colour in the main menu when return from setting -> music volume
<img width="1280" height="1024" alt="Image" src="https://github.com/user-attachments/assets/5322a3d0-bc27-495c-a518-3241217d75b1" />

The game does not look properly scale it (i prefer lost aspect ratio when fit to entire window just is correctly rendered)
<img width="1280" height="1024" alt="Image" src="https://github.com/user-attachments/assets/7bf324e8-cf6f-4171-acf3-57f09e7442bb" />
<img width="1280" height="1024" alt="Image" src="https://github.com/user-attachments/assets/40c640b7-a336-433e-bafb-b34339edc7dd" />
<img width="1280" height="1024" alt="Image" src="https://github.com/user-attachments/assets/476083e1-551d-4740-8f8a-1dcb3920505c" />
<img width="1280" height="1024" alt="Image" src="https://github.com/user-attachments/assets/7915f7d3-d2b3-4ddf-a464-8c4c85449e15" />

With the rockdroid 2 game : i notice not have the same issue with menu text and area with different colors when i return to main menu. but the image scale inside the window looks like the same

<img width="1280" height="1024" alt="Image" src="https://github.com/user-attachments/assets/baddc644-86f1-423f-afbf-c19c51540da8" />

well the rockdroid 2 output on cli : 
[ inukaze  | 14-11-2025 | 05:24 pm ]
[build]$ `./rockbot `

> Renderer backend: opengl
> SDL version: 2.0.20
> Error in soundLib::load_boss_music::Mix_LoadMUS('/tmp/rockbot/build/games/RockDroid2/music/rockbot_boss.mod': 'Unrecognized audio format'
> Error in soundLib::load_music::Mix_LoadMUS('/tmp/rockbot/build/games/RockDroid2/music/intro_scene.mod': 'Unrecognized audio format'
> ERROR: play_music, music is null
> FIO:_STRINGS::get_string_list_from_scene_text_file[/tmp/rockbot/build//games/RockDroid2/scenes/text/es/0.txt]
> font-size[8], max_line_w[17], adjust_x[64], center_x[96]
> Renderer backend: opengl
> SDL version: 2.0.20
> Renderer backend: opengl
> SDL version: 2.0.20
> Error in soundLib::load_music::Mix_LoadMUS('/tmp/rockbot/build/games/RockDroid2/music/rockbot_2_open_final.mod': 'Unrecognized audio format'
> ERROR: play_music, music is null
> RESET cheat input - expected-button[0], pressed-button[1]
> RESET cheat input - expected-button[0], pressed-button[1]
> RESET cheat input - expected-button[0], pressed-button[1]
> CONTINUE cheat input
> RESET cheat input - expected-button[0], pressed-button[7]
> CONTINUE cheat input
> CONTINUE cheat input
> RESET cheat input - expected-button[1], pressed-button[7]
> Error in soundLib::load_music::Mix_LoadMUS('/tmp/rockbot/build/games/RockDroid2/music/porto_alegre.mod': 'Unrecognized audio format'
> >>>>> CLASSNPC.BUILD[OCTOPUS] - start_point[1][0], pos[134][112], bg_pos.x[118]
> ERROR: play_music, music is null
> ### CHARACTER::REMOVE PLATFORM #2
> ### JUMP::START - bigjump_mode[0]
> ### CHARACTER::REMOVE PLATFORM #2
> ### JUMP::START - bigjump_mode[0]
> ### CHARACTER::REMOVE PLATFORM #2
> ### JUMP::START - bigjump_mode[0]
> ### CHARACTER::REMOVE PLATFORM #2
> ### JUMP::START - bigjump_mode[0]
> ### CHARACTER::REMOVE PLATFORM #2
> ### JUMP::START - bigjump_mode[0]
> ### CHARACTER::REMOVE PLATFORM #2
> ### JUMP::START - bigjump_mode[0]
> ### CHARACTER::REMOVE PLATFORM #2
> ### JUMP::START - bigjump_mode[0]
> ### CHARACTER::REMOVE PLATFORM #2
> ### JUMP::START - bigjump_mode[0]
> ### CHARACTER::REMOVE PLATFORM #2
> ### JUMP::START - bigjump_mode[0]
> ### CHARACTER::REMOVE PLATFORM #2
> ### JUMP::START - bigjump_mode[0]
> ### CHARACTER::REMOVE PLATFORM #2
> ### JUMP::START - bigjump_mode[0]
> ### CHARACTER::REMOVE PLATFORM #2
> ### JUMP::START - bigjump_mode[0]
> ### CHARACTER::REMOVE PLATFORM #2
> ### JUMP::START - bigjump_mode[0]
> ### CHARACTER::REMOVE PLATFORM #2
> ### JUMP::START - bigjump_mode[0]
> 
>

**Comment by humbertodias** [comment]
@inukaze 
Got it!
Based on your output:

```
SDL version: 2.0.20
Error in soundLib::load_boss_music::Mix_LoadMUS('/tmp/rockbot/build/games/RockDroid2/music/rockbot_boss.mod': 'Unrecognized audio format')
```

It seems your `SDL2_mixer` library wasn’t compiled with support for `Mix_LoadMUS(MOD_MODPLUG)`, which is required to play `.mod` files.

Please check whether your `libSDL2_mixer` contains the necessary symbols:

```bash
strings /usr/lib/libSDL2_mixer.so | grep -i mod
```

If nothing is returned, that confirms the issue.

As a workaround, I recommend installing a version of SDL2_mixer that includes `Mix_LoadMUS` via your package manager (`slapt-get`) or recompiling it from source.

You can try rebuilding SDL2_mixer with the needed modules enabled:

```bash
wget https://github.com/libsdl-org/SDL_mixer/releases/download/release-2.8.0/SDL2_mixer-2.8.0.tar.gz
tar xvf SDL2_mixer-2.8.0.tar.gz
cd SDL2_mixer-2.8.0/

cmake -B build \
  -DCMAKE_BUILD_TYPE=Debug \
  -DCMAKE_INSTALL_PREFIX=/usr \
  -DSDL2MIXER_VORBIS=VORBISFILE \
  -DSDL2MIXER_MOD_MODPLUG=ON \
  -DSDL2MIXER_MP3_MPG123=ON \
  -DSDL2MIXER_FLAC=ON \
  -DSDL2MIXER_OPUS=ON \
  -DSDL2MIXER_MIDI_FLUIDSYNTH=OFF \
  -DSDL2MIXER_MIDI_NATIVE=OFF \
  -DSDL2MIXER_WAVPACK=OFF

cmake --build build
sudo make install
```
This should generate `libSDL2_mixer-2.0d.so` and enabling `-DSDL2MIXER_MOD_MODPLUG=ON` will add support for the following module file types: `.mod, .xm, .it, and .s3m`.

```
[ 89%] Linking C shared library libSDL2_mixer-2.0d.so
[ 89%] Built target SDL2_mixer
[ 92%] Building C object CMakeFiles/playmus.dir/playmus.c.o
[ 94%] Linking C executable playmus
[ 94%] Built target playmus
[ 97%] Building C object CMakeFiles/playwave.dir/playwave.c.o
[100%] Linking C executable playwave
[100%] Built target playwave
```
Now, it must contains the `Mix_LoadMUS`
```
strings build/libSDL2_mixer-2.0d.so | grep -i mod
```
output
```
music_modplug.c
modplug
MODPLUG_Load
MODPLUG_Unload
MODPLUG_Open
```

If that doesn't work. it should be related to `pulseaudio`

Please give it a try and let us know how it goes. Thanks!

**Comment by inukaze** [comment]
> [@inukaze](https://github.com/inukaze) Got it! Based on your output:
> 
> ```
> SDL version: 2.0.20
> Error in soundLib::load_boss_music::Mix_LoadMUS('/tmp/rockbot/build/games/RockDroid2/music/rockbot_boss.mod': 'Unrecognized audio format')
> ```
> 
> It seems your `SDL2_mixer` library wasn’t compiled with support for `Mix_LoadMUS(MOD_MODPLUG)`, which is required to play `.mod` files.
> 
> Please check whether your `libSDL2_mixer` contains the necessary symbols:
> 
> strings /usr/lib/libSDL2_mixer.so | grep -i mod
> If nothing is returned, that confirms the issue.
> 
> As a workaround, I recommend installing a version of SDL2_mixer that includes `Mix_LoadMUS` via your package manager (`slapt-get`) or recompiling it from source.
> 
> You can try rebuilding SDL2_mixer with the needed modules enabled:
> 
> wget https://github.com/libsdl-org/SDL_mixer/releases/download/release-2.8.0/SDL2_mixer-2.8.0.tar.gz
> tar xvf SDL2_mixer-2.8.0.tar.gz
> cd SDL2_mixer-2.8.0/
> 
> cmake -B build \
>   -DCMAKE_BUILD_TYPE=Debug \
>   -DCMAKE_INSTALL_PREFIX=/usr \
>   -DSDL2MIXER_VORBIS=VORBISFILE \
>   -DSDL2MIXER_MOD_MODPLUG=ON \
>   -DSDL2MIXER_MP3_MPG123=ON \
>   -DSDL2MIXER_FLAC=ON \
>   -DSDL2MIXER_OPUS=ON \
>   -DSDL2MIXER_MIDI_FLUIDSYNTH=OFF \
>   -DSDL2MIXER_MIDI_NATIVE=OFF \
>   -DSDL2MIXER_WAVPACK=OFF
> 
> cmake --build build
> sudo make install
> This should generate `libSDL2_mixer-2.0d.so` and enabling `-DSDL2MIXER_MOD_MODPLUG=ON` will add support for the following module file types: `.mod, .xm, .it, and .s3m`.
> 
> ```
> [ 89%] Linking C shared library libSDL2_mixer-2.0d.so
> [ 89%] Built target SDL2_mixer
> [ 92%] Building C object CMakeFiles/playmus.dir/playmus.c.o
> [ 94%] Linking C executable playmus
> [ 94%] Built target playmus
> [ 97%] Building C object CMakeFiles/playwave.dir/playwave.c.o
> [100%] Linking C executable playwave
> [100%] Built target playwave
> ```
> 
> Now, it must contains the `Mix_LoadMUS`
> 
> ```
> strings build/libSDL2_mixer-2.0d.so | grep -i mod
> ```
> 
> output
> 
> ```
> music_modplug.c
> modplug
> MODPLUG_Load
> MODPLUG_Unload
> MODPLUG_Open
> ```
> 
> If that doesn't work. it should be related to `pulseaudio`
> 
> Please give it a try and let us know how it goes. Thanks!

`strings /usr/lib/libSDL2_mixer.so | grep -i mod`

> MOD support not available

```
echo -ne 'GLIBC Versions Supported by "'This GNU/Linux Distribution'" :\n\n' ; strings /lib64/libc.so.6 | grep -E "^GLIBC_.{3,4}$" | sort -V | uniq 
GLIBC Versions Supported by "This GNU/Linux Distribution" :

```

> GLIBC_2.3
> GLIBC_2.4
> GLIBC_2.5
> GLIBC_2.6
> GLIBC_2.7
> GLIBC_2.8
> GLIBC_2.9
> GLIBC_2.10
> GLIBC_2.11
> GLIBC_2.12
> GLIBC_2.13
> GLIBC_2.14
> GLIBC_2.15
> GLIBC_2.16
> GLIBC_2.17
> GLIBC_2.18
> GLIBC_2.22
> GLIBC_2.23
> GLIBC_2.24
> GLIBC_2.25
> GLIBC_2.26
> GLIBC_2.27
> GLIBC_2.28
> GLIBC_2.29
> GLIBC_2.30
> GLIBC_2.32
> GLIBC_2.33
> 

root[build]# `slpkg update ; slpkg -F SDL2`

> 
> Check and update repositories:
> 
> Check repository [slack] ... 
> [1/1][ Download ] --> PACKAGES.TXT
> 
> Will not apply HSTS. The HSTS database must be a regular and non-world-writable file.
> ERROR: could not open HSTS store at '/root/.wget-hsts'. HSTS will be disabled.
> --2025-11-14 21:35:43--  http://mirrors.slackware.com/slackware/slackware64-15.0/PACKAGES.TXT
> Resolviendo mirrors.slackware.com (mirrors.slackware.com)... 207.223.116.213
> Conectando con mirrors.slackware.com (mirrors.slackware.com)[207.223.116.213]:80... conectado.
> Petición HTTP enviada, esperando respuesta... 302 Found
> Localización: https://mirrors.slackware.com/slackware/slackware64-15.0/PACKAGES.TXT [siguiendo]
> --2025-11-14 21:35:45--  https://mirrors.slackware.com/slackware/slackware64-15.0/PACKAGES.TXT
> Conectando con mirrors.slackware.com (mirrors.slackware.com)[207.223.116.213]:443... conectado.
> Petición HTTP enviada, esperando respuesta... 302 Found
> Localización: https://mirror.cedia.org.ec/slackware/slackware64-15.0/PACKAGES.TXT [siguiendo]
> --2025-11-14 21:35:46--  https://mirror.cedia.org.ec/slackware/slackware64-15.0/PACKAGES.TXT
> Resolviendo mirror.cedia.org.ec (mirror.cedia.org.ec)... 201.159.221.67, 2800:68:0:bebe::3
> Conectando con mirror.cedia.org.ec (mirror.cedia.org.ec)[201.159.221.67]:443... conectado.
> Petición HTTP enviada, esperando respuesta... 200 OK
> Longitud: 864745 (844K) [text/plain]
> Grabando a: «/var/lib/slpkg/slack_repo/core/PACKAGES.TXT»
> 
> PACKAGES.TXT            45%[===========>               ] 383,75K  --.-KB/s    en 15m 16s 
> 
> 2025-11-14 21:51:13 (429 B/s) - Error de lectura en el byte 392964/864745 (Expiró el tiempo de conexión). Reintentando.
> 
> --2025-11-14 21:51:14--  (intento: 2)  https://mirror.cedia.org.ec/slackware/slackware64-15.0/PACKAGES.TXT
> Conectando con mirror.cedia.org.ec (mirror.cedia.org.ec)[201.159.221.67]:443... conectado.
> Petición HTTP enviada, esperando respuesta... 206 Partial Content
> Longitud: 864745 (844K), quedan 471781 (461K) [text/plain]
> Grabando a: «/var/lib/slpkg/slack_repo/core/PACKAGES.TXT»
> 
> PACKAGES.TXT           100%[++++++++++++==============>] 844,48K  20,6KB/s    en 20s     
> 
> 2025-11-14 21:51:35 (22,7 KB/s) - «/var/lib/slpkg/slack_repo/core/PACKAGES.TXT» guardado [864745/864745]
> 
> 
> [1/1][ Download ] --> CHECKSUMS.md5
> 
> Will not apply HSTS. The HSTS database must be a regular and non-world-writable file.
> ERROR: could not open HSTS store at '/root/.wget-hsts'. HSTS will be disabled.
> --2025-11-14 21:51:35--  http://mirrors.slackware.com/slackware/slackware64-15.0/CHECKSUMS.md5
> Resolviendo mirrors.slackware.com (mirrors.slackware.com)... 207.223.116.213
> Conectando con mirrors.slackware.com (mirrors.slackware.com)[207.223.116.213]:80... conectado.
> Petición HTTP enviada, esperando respuesta... 302 Found
> Localización: https://mirrors.slackware.com/slackware/slackware64-15.0/CHECKSUMS.md5 [siguiendo]
> --2025-11-14 21:51:36--  https://mirrors.slackware.com/slackware/slackware64-15.0/CHECKSUMS.md5
> Conectando con mirrors.slackware.com (mirrors.slackware.com)[207.223.116.213]:443... conectado.
> Petición HTTP enviada, esperando respuesta... 200 OK
> Longitud: 1432104 (1,4M) [text/plain]
> Grabando a: «/var/lib/slpkg/slack_repo/core/CHECKSUMS.md5»
> 
> CHECKSUMS.md5          100%[==========================>]   1,37M  25,9KB/s    en 61s     
> 
> 2025-11-14 21:52:39 (23,1 KB/s) - «/var/lib/slpkg/slack_repo/core/CHECKSUMS.md5» guardado [1432104/1432104]
> 
> 
> [1/1][ Download ] --> PACKAGES.TXT
> 
> Will not apply HSTS. The HSTS database must be a regular and non-world-writable file.
> ERROR: could not open HSTS store at '/root/.wget-hsts'. HSTS will be disabled.
> --2025-11-14 21:52:39--  http://mirrors.slackware.com/slackware/slackware64-15.0/extra/PACKAGES.TXT
> Resolviendo mirrors.slackware.com (mirrors.slackware.com)... 207.223.116.213
> Conectando con mirrors.slackware.com (mirrors.slackware.com)[207.223.116.213]:80... conectado.
> Petición HTTP enviada, esperando respuesta... 302 Found
> Localización: https://mirrors.slackware.com/slackware/slackware64-15.0/extra/PACKAGES.TXT [siguiendo]
> --2025-11-14 21:52:39--  https://mirrors.slackware.com/slackware/slackware64-15.0/extra/PACKAGES.TXT
> Conectando con mirrors.slackware.com (mirrors.slackware.com)[207.223.116.213]:443... conectado.
> Petición HTTP enviada, esperando respuesta... 302 Found
> Localización: https://mirror.cedia.org.ec/slackware/slackware64-15.0/extra/PACKAGES.TXT [siguiendo]
> --2025-11-14 21:52:40--  https://mirror.cedia.org.ec/slackware/slackware64-15.0/extra/PACKAGES.TXT
> Resolviendo mirror.cedia.org.ec (mirror.cedia.org.ec)... 201.159.221.67, 2800:68:0:bebe::3
> Conectando con mirror.cedia.org.ec (mirror.cedia.org.ec)[201.159.221.67]:443... conectado.
> Petición HTTP enviada, esperando respuesta... 200 OK
> Longitud: 40404 (39K) [text/plain]
> Grabando a: «/var/lib/slpkg/slack_repo/extra/PACKAGES.TXT»
> 
> PACKAGES.TXT           100%[==========================>]  39,46K  40,9KB/s    en 1,0s    
> 
> 2025-11-14 21:52:42 (40,9 KB/s) - «/var/lib/slpkg/slack_repo/extra/PACKAGES.TXT» guardado [40404/40404]
> 
> 
> [1/1][ Download ] --> CHECKSUMS.md5
> 
> Will not apply HSTS. The HSTS database must be a regular and non-world-writable file.
> ERROR: could not open HSTS store at '/root/.wget-hsts'. HSTS will be disabled.
> --2025-11-14 21:52:42--  http://mirrors.slackware.com/slackware/slackware64-15.0/extra/CHECKSUMS.md5
> Resolviendo mirrors.slackware.com (mirrors.slackware.com)... 207.223.116.213
> Conectando con mirrors.slackware.com (mirrors.slackware.com)[207.223.116.213]:80... conectado.
> Petición HTTP enviada, esperando respuesta... 302 Found
> Localización: https://mirrors.slackware.com/slackware/slackware64-15.0/extra/CHECKSUMS.md5 [siguiendo]
> --2025-11-14 21:52:42--  https://mirrors.slackware.com/slackware/slackware64-15.0/extra/CHECKSUMS.md5
> Conectando con mirrors.slackware.com (mirrors.slackware.com)[207.223.116.213]:443... conectado.
> Petición HTTP enviada, esperando respuesta... 200 OK
> Longitud: 60332 (59K) [text/plain]
> Grabando a: «/var/lib/slpkg/slack_repo/extra/CHECKSUMS.md5»
> 
> CHECKSUMS.md5          100%[==========================>]  58,92K  14,6KB/s    en 4,0s    
> 
> 2025-11-14 21:52:47 (14,6 KB/s) - «/var/lib/slpkg/slack_repo/extra/CHECKSUMS.md5» guardado [60332/60332]
> 
> 
> [1/1][ Download ] --> ChangeLog.txt
> 
> Will not apply HSTS. The HSTS database must be a regular and non-world-writable file.
> ERROR: could not open HSTS store at '/root/.wget-hsts'. HSTS will be disabled.
> --2025-11-14 21:52:47--  http://mirrors.slackware.com/slackware/slackware64-15.0/ChangeLog.txt
> Resolviendo mirrors.slackware.com (mirrors.slackware.com)... 207.223.116.213
> Conectando con mirrors.slackware.com (mirrors.slackware.com)[207.223.116.213]:80... conectado.
> Petición HTTP enviada, esperando respuesta... 302 Found
> Localización: https://mirrors.slackware.com/slackware/slackware64-15.0/ChangeLog.txt [siguiendo]
> --2025-11-14 21:52:47--  https://mirrors.slackware.com/slackware/slackware64-15.0/ChangeLog.txt
> Conectando con mirrors.slackware.com (mirrors.slackware.com)[207.223.116.213]:443... conectado.
> Petición HTTP enviada, esperando respuesta... 200 OK
> Longitud: 2273479 (2,2M) [text/plain]
> Grabando a: «/var/log/slpkg/slack/ChangeLog.txt»
> 
> ChangeLog.txt          100%[==========================>]   2,17M  21,3KB/s    en 1m 42s  
> 
> 2025-11-14 21:54:30 (21,8 KB/s) - «/var/log/slpkg/slack/ChangeLog.txt» guardado [2273479/2273479]
> 
> Done
> Check repository [sbo] ... Done
> Check repository [alien] ... ^C
> Packages with name matching [ SDL2 ]
> 
> +==============================================================================
> | Repository Package                                                       Size
> +==============================================================================
>   slack      SDL2-2.0.20-x86_64-1.txz                                     804 K
>   slack      SDL2_gfx-1.0.4-x86_64-3.txz                                   32 K
>   slack      SDL2_image-2.0.5-x86_64-3.txz                                 64 K
>   slack      SDL2_mixer-2.0.4-x86_64-5.txz                                 68 K
>   slack      SDL2_net-2.0.1-x86_64-4.txz                                   16 K
>   slack      SDL2_ttf-2.0.18-x86_64-1.txz                                  32 K
>   sbo        SDL2_Pango-2.1.5                                               0 K
>   sbo        SDL2_sound-2.0.4                                               0 K
>   slonly     PySDL2-0.9.5-x86_64-1_slonly.txz                             384 K
>   multi      SDL2-compat32-2.0.20-x86_64-1compat32.txz                    540 K
>   multi      SDL2_gfx-compat32-1.0.4-x86_64-3compat32.txz                  28 K
>   multi      SDL2_image-compat32-2.0.5-x86_64-3compat32.txz                60 K
>   multi      SDL2_mixer-compat32-2.0.4-x86_64-5compat32.txz                60 K
>   multi      SDL2_net-compat32-2.0.1-x86_64-4compat32.txz                  12 K
>   multi      SDL2_ttf-compat32-2.0.18-x86_64-1compat32.txz                 24 K
> 
> Found summary
> ===============================================================================
> Total found 15 packages in 4 repositories.
> 


Well after i see if i wrote a custom SlackBuild for compile SDL2 with Mixer because on SlackBuilds.org i just got : 
https://slackbuilds.org/repository/15.0/libraries/SDL2_sound/

But is not the same thing like SDL2_mixer

The is the source code of that SlackBuild ( an automated script for download, compile, and create slackware package)
https://slackbuilds.org/slackbuilds/15.0/libraries/SDL2_sound/SDL2_sound.SlackBuild

**Comment by humbertodias** [comment]
@inukaze 
Terrific! So, did it work now?

**Comment by protoman** [comment]
1 - Why is the music in "mod" format?
The idea is to keep the app small, and use less RAM memory. Also, it was easier to find chiptunes in this format.

2 - Do you plan to make a version for MS-DOS 5.22?
That would be fun, but I did left DOS ages ago. I do plan a PS3 port altough.

3 - Does the music format only support mod, not "it" or "xm"?
It does support it and XM
